ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -929,9 +941,56 @@ var _ = Context("Inside the default namespace", func() {
 
 		It("Check the number of worker Pods", func() {
 			numWorkerPods := 3 * int(numOfHosts)
+			desiredWorkerPods := replicas * numOfHosts
+			minWorkerPods := minReplicas * numOfHosts
+			maxWorkerPods := maxReplicas * numOfHosts
+
 			Eventually(
 				listResourceFunc(ctx, &workerPods, workerFilters...),
 				time.Second*3, time.Millisecond*500).Should(Equal(numWorkerPods), fmt.Sprintf("workerGroup %v", workerPods.Items))
+
+			Eventually(func() bool {
```

Comment:
Move the logic to "RayCluster's .status.state should be updated to 'ready' shortly after all Pods are Running"